src/averageAges.js

```diff
@@ -55,7 +73,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let kids = people.filter(kid => {
+    return people.some(mom => mom.name === kid.mother);
+  });
+
+  onlyWithSon && (kids = kids.filter(kid => kid.sex === 'm'));
+
+  const ages = kids.map(kid => {
+    const momObj = people.find(mom => mom.name === kid.mother);
+
+    return kid.born - momObj.born;
+  });
+
+  return ages.reduce(reducer) / ages.length;
```

Comment:
```suggestion
  const mothers = people.reduce((acc, person) => {
    acc[person.name] = person;

    return acc;
  }, {});

  const { sum, childrenCount } = people.reduce((acc, person) => {
    const { sex, mother, born } = person;

    const isPerson = onlyWithSon
      ? sex === 'm' && mothers[mother]
      : mothers[mother];

    return isPerson ? {
      sum: acc.sum + born - mothers[mother].born,
      childrenCount: acc.childrenCount + 1,
    } : acc;
  }, {
    sum: 0, childrenCount: 0,
  });

  return sum / childrenCount;
```

Reply:
It is nor required, I just wanted to show how we can improve performance and make just two itearation on array.